### PR TITLE
BellmanFord 

### PR DESCRIPTION
**Pull request by YanisLcn**

### CODE REVIEW

**Review by Yag000 — CHANGES_REQUESTED** [review]
It has not been added to the notebook itself, it won't be visible on the pdf

File: content/graph/BellmanFord.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+"""
+Auhtor: Unknown
+Source: https://en.wikipedia.org/wiki/Bellman%E2%80%93Ford_algorithm
+Description: Computes shortest paths from $start$. Allow negatives-weight cycles
```
Comment (Yag000) [inline content/graph/BellmanFord.py]:
```suggestion
Description: Computes shortest paths from $start$. Allows negative-weighted cycles
```

**Review by Yag000 — CHANGES_REQUESTED** [review]

**Comment by Yag000** [comment]
> It has not been added to the notebook itself, it won't be visible on the pdf

You can see how it can be achieved [here](https://github.com/UPCGM-ICPC/UPCGM-Notebook/pull/4/files).

File: content/graph/BellmanFord.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
+"""
+Auhtor: Unknown
+Source: https://en.wikipedia.org/wiki/Bellman%E2%80%93Ford_algorithm
+Description: Computes shortest paths from $start$. Allows negative-weighted cycles
+Time: $O(VE)$
```
Comment (Yag000) [inline content/graph/BellmanFord.py]:
```suggestion
* Auhtor: Unknown
* Source: https://en.wikipedia.org/wiki/Bellman%E2%80%93Ford_algorithm
* Description: Computes shortest paths from $start$. Allows negative-weighted cycles
* Time: $O(VE)$
```
Won't compile without this

**Review by Yag000 — CHANGES_REQUESTED** [review]

File: content/graph/BellmanFord.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
+"""
+* Auhtor: Unknown
+* Source: https://en.wikipedia.org/wiki/Bellman%E2%80%93Ford_algorithm
+* Description: Computes shortest paths from $start$. Allows negative-weighted cycles
+* Time: $O(VE)$
```
Comment (Yag000) [inline content/graph/BellmanFord.py]:
```suggestion
* Time: O(VE)
```

This might be the issue

Comment (YanisLcn) [reply content/graph/BellmanFord.py]: I think the issue is caused by 'Auhtor' instead of 'Author'

File: content/graph/BellmanFord.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
+"""
+ * Author: Unknown
+ * Source: https://en.wikipedia.org/wiki/Bellman%E2%80%93Ford_algorithm
+ * Description: Computes shortest paths from $start$. Allows negative-weighted cycles
+ * Time: $O(VE)$
+"""
+
+
+def bellmanFord(graph, start_vertex):
```
Comment (Yag000) [inline content/graph/BellmanFord.py:7]: This function does not return anything

File: content/graph/BellmanFord.py
Comment (Yag000) [inline content/graph/BellmanFord.py:1]: the code does not "compile". Using the correct data structure, `Graph`, and adding this code:
```python
if __name__ == "__main__":
    graph = Graph(5, [(0, 1, 1), (1, 2, 1), (2, 3, 1), (3, 1, -3), (1, 4, 1)])
    bellmanFord(graph, 0)
```
yields this error:
```
Traceback (most recent call last):
  File "/home/yago/Projects/UPCGM-Notebook/content/graph/BellmanFord.py", line 63, in <module>
    bellmanFord(graph, 0)
  File "/home/yago/Projects/UPCGM-Notebook/content/graph/BellmanFord.py", line 46, in bellmanFord
    for _ in range(len(graph.v) - 1):
                   ^^^^^^^^^^^^
TypeError: object of type 'int' has no len()
```

Comment (Yag000) [reply content/graph/BellmanFord.py:1]: Furthermore, fixing that error, yields this result:
```
Traceback (most recent call last):
  File "/home/yago/Projects/UPCGM-Notebook/content/graph/BellmanFord.py", line 63, in <module>
    bellmanFord(graph, 0)
  File "/home/yago/Projects/UPCGM-Notebook/content/graph/BellmanFord.py", line 47, in bellmanFord
    for u, v, w in graph.edges:
        ^^^^^^^
TypeError: cannot unpack non-iterable int object
```

**Review by Yag000 — CHANGES_REQUESTED** [review]
the code does not work. Testing the code is always welcomed :)

File: content/graph/BellmanFord.py
Comment (YanisLcn) [reply content/graph/BellmanFord.py:1]: Added method for that

File: content/graph/GraphClass.py [hunk 5]
```diff
@@ -27,3 +27,10 @@ def get_edges(self):
             for v in self.edges[u].keys():
                 edges.append((u, v))
         return edges
+
```
Comment (Yag000) [inline content/graph/GraphClass.py]: Could you remove this empty line

**Review by Yag000 — CHANGES_REQUESTED** [review]

**Comment by Yag000** [comment]
Using the same example as [previouly](https://github.com/UPCGM-ICPC/UPCGM-Notebook/pull/5#discussion_r1461052745) I get a completely false result... 

```py
if __name__ == "__main__":
    graph = Graph(5, [(0, 1, 1), (1, 2, 1), (2, 3, 1), (3, 1, -3), (1, 4, 1)])
    print(bellmanFord(graph, 0))
```

result:
```
[0, -inf, -inf, -inf, -inf]
```

**Review by Yag000 — APPROVED** [review]
LGTM

**Pull request merged**